tao_compiler/mlir/disc/transforms/lhlo_elemental_utils.cc

```diff
@@ -104,12 +122,20 @@ Value createLoadOrUseCachedValue(Location loc, OpBuilder* b, Operation* op,
 
 Value mayCreateStore(OpBuilder* b, Location loc, Operation* op, Value value,
                      ValueRange out_indices, LowerConfig* lower_config) {
-  if (lower_config != nullptr && lower_config->isWrittenBack(op)) {
+  if (lower_config != nullptr) {
     Value out_memref = cast<lmhlo::LmhloOp>(op).getResultBuffer();
     if (out_memref == nullptr) {
       return nullptr;
     }
-    b->create<memref::StoreOp>(loc, value, out_memref, out_indices);
+    // Write back to off-chip memory.
+    if (lower_config->isWrittenBack(op)) {
```

Comment:
would also write to shared mem if `isWrittenBack(op)`?

Reply:
They are different. `isWrittenBack` is to write back to global memory, not shared memory.